README.md

```diff
@@ -211,6 +211,49 @@ This mixin binds the url of the best fitting image to the source attribute, base
 
 This mixin binds the url of the best fitting image as the background url to the elements style attribute, based in the values provided by the `image` and `size` attribute. It also get the `responsiveImage` service injected.
 
+##Extensibility hooks
+### Extend the image processing
+
+During the image process the addon calls the `preProcessImage` and the `postProcessImage` hooks for each origin image and supported width. Here you can add custom image process steps, like a watermark integration. The first hook will be called just before the addon's image process calls applies, the latter after. You can register your callbacks by calling the addon's `addImagePreProcessor` or `addImagePostProcessor` function before the addon's `postprocessTree` was called.
+In both cases the callback method you provide must have the following signature:
+
+```javascript
+  function preProcessor(sharp, image, width, configuration)
+  {
+    return sharp;
+  }
+```
+* **sharp:** [sharp](https://github.com/lovell/sharp) object with the current origin image
+* **image:** the name of the origin image file
+* **width:** the width of the resulting resized image of the current process
+* **configuration:** the configuration for the current image processing (from environments configuration)
+
+The callback must return a `sharp`-object or a Promise resolves to it.
+
+**Note:** In addition to the callback, you can also pass an optional target object that will be set as `this` on the context. This is a good way to give your function access to the current object.
+
+For an example see [ember-lazy-responsive-image](https://github.com/kaliber5/ember-lazy-responsive-image/blob/master/index.js)
+### Extend the metadata
+
+Before the addon injects the generated metadata into the build, a `extendMetadata`-hook is called for each origin image. Here you can add custom metadata.  You can register your callbacks by calling the addon's `addMetadataExtension` function before the addon's `postprocessTree` was called.
```

Comment:
Maybe add a sentence for context, that this metadata is what is returned by the service's `getImages ` and `getImageDataBySize` methods?

Reply:
I added a sentence and also point to the `ember-lazy-responsive-image`-addon as an example. Do you think it's enough?

Reply:
Sure